src/Avalonia.Controls.TreeDataGrid/Primitives/TreeDataGridCell.cs

```diff
@@ -153,11 +163,27 @@ protected override void OnKeyDown(KeyEventArgs e)
         {
             base.OnKeyDown(e);
 
-            if (!_isEditing && CanEdit && !e.Handled && e.Key == Key.F2)
+            if (Model is null || e.Handled)
+                return;
+
+            if (e.Key == Key.F2 && 
+                !IsEditing && 
+                Model.CanEdit &&
+                IsEnabledEditGesture(EditGestures.F2, Model.EditGestures))
             {
                 BeginEdit();
                 e.Handled = true;
             }
+            else if (e.Key == Key.Enter && IsEditing)
```

Comment:
I think it makes some sense to rename EditGestures to BeginEditGestures and add EndEditGestures.
But that's minor.

Reply:
Done.